service/create/service.go

```diff
@@ -492,10 +519,21 @@ func (s *Service) runMachine(input runMachineInput) error {
 			},
 		},
 	}); err != nil {
-		return microerror.MaskAny(err)
+		return nil, microerror.MaskAny(err)
 	}
 
 	s.logger.Log("info", fmt.Sprintf("instance '%s' tagged", input.name))
+	instance := reservation.Instances[0]
 
-	return nil
+	return instance, nil
```

Comment:
Define inefficient. We do not have performance problems at the moment. We are in a "make it work" phase. For that to happen we should try to get the design right. I know that this here is no big deal but based on my personal experience such things add up quite quickly and then you have a codebase that is hard to maintain. I would like to prevent this and get the design right. For my understanding, how can we fetch instance information? How do we need to reference an instance to fetch information about it from EC2? Imagine we would only return an error when creating the instances. How would it look like to fetch information that we then need?